systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.upgrade;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.enums.OlmInstallationStrategy;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBasicExampleClients;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBridgeExampleClients;
+import io.strimzi.systemtest.resources.operator.OlmResource;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.FileUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.DeploymentUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.StatefulSetUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import static io.strimzi.systemtest.resources.ResourceManager.cmdKubeClient;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+
+public class OlmUpgradeST extends AbstractUpgradeST {
+
+    private static final Logger LOGGER = LogManager.getLogger(OlmUpgradeST.class);
+
+    private final String namespace = "olm-upgrade-namespace";
+    private final String producerName = "producer";
+    private final String consumerName = "consumer";
+    private final String topicUpgradeName = "topic-upgrade";
+    private final int messageUpgradeCount =  10_000;
+    private final Map<String, List<String>> mapOfKafkaVersionsWithSupportedClusterOperators = getMapKafkaVersionsWithSupportedClusterOperatorVersions();
+
+    @Test
+    void testUpgrade() {
+        Map<String, String> kafkaSnapshot = null;
+        boolean isUpgradeAble = true;
+
+        while (isUpgradeAble) {
```

Comment:
Wonder if wouldn't be better to use `StrimziUpgradeST.json` for each upgrade phase as we do in StrimziUpgradeST to avoid mistakes during manfiests image build. Imagine, that manifest image iwll does not contain latest manifests cause some issue during build. The tests will pass anyway, because it will not care if latest version should be X or Y. WDYT?

Reply:
It's done!